kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/jsinterop/util/ExpressionFiller.java

```diff
@@ -0,0 +1,596 @@
+            final DecisionTable decisionTableExpression = (DecisionTable) wrappedExpression;
+            final String hitPolicy = decisionTableExpression.getHitPolicy() != null ? decisionTableExpression.getHitPolicy().value() : null;
+            final String aggregation = decisionTableExpression.getAggregation() != null ? decisionTableExpression.getAggregation().getCode() : "";
+            return new DecisionTableProps(expressionName, dataType, hitPolicy, aggregation,
+                                          annotationsConvertForDecisionTableProps(decisionTableExpression),
+                                          inputConvertForDecisionTableProps(decisionTableExpression),
+                                          outputConvertForDecisionTableProps(decisionTableExpression),
+                                          rulesConvertForDecisionTableProps(decisionTableExpression));
+        }
+        return new ExpressionProps(expressionName, dataType, null);
+    }
+
+    /**
+     * MODEL filling
+     */
+
+    private static Expression buildAndFillNestedExpression(final ExpressionProps props) {
+        if (LITERAL_EXPRESSION.getText().equals(props.logicType)) {
+            final LiteralExpression literalExpression = new LiteralExpression();
+            fillLiteralExpression(literalExpression, (LiteralExpressionProps) props);
+            return literalExpression;
+        } else if (CONTEXT.getText().equals(props.logicType)) {
+            final Context contextExpression = new Context();
+            fillContextExpression(contextExpression, (ContextProps) props);
+            return contextExpression;
+        } else if (RELATION.getText().equals(props.logicType)) {
+            final Relation relationExpression = new Relation();
+            fillRelationExpression(relationExpression, (RelationProps) props);
+            return relationExpression;
+        } else if (LIST.getText().equals(props.logicType)) {
+            final List listExpression = new List();
+            fillListExpression(listExpression, (ListProps) props);
+            return listExpression;
+        } else if (INVOCATION.getText().equals(props.logicType)) {
+            final Invocation invocationExpression = new Invocation();
+            fillInvocationExpression(invocationExpression, (InvocationProps) props);
+            return invocationExpression;
+        } else if (FUNCTION.getText().equals(props.logicType)) {
+            final FunctionDefinition functionExpression = new FunctionDefinition();
+            fillFunctionExpression(functionExpression, (FunctionProps) props);
+            return functionExpression;
+        } else if (DECISION_TABLE.getText().equals(props.logicType)) {
+            final DecisionTable decisionTableExpression = new DecisionTable();
+            fillDecisionTableExpression(decisionTableExpression, (DecisionTableProps) props);
+            return decisionTableExpression;
+        }
+        return null;
```

Comment:
Shouldn't we throw some exception about illegal state?

Reply:
In that case, we are filling an empty nested expression model. I've tried, and it seems that when the nested expression is not set, we should set `null` for it. So, I would say it is a normal flow that an exception should not block.